src/action/network.rs

```diff
@@ -0,0 +1,296 @@
+            local_address: Some(make_network_endpoint(
+                &udp_info.local_addr,
+                udp_info.local_port
+            )),
+            remote_address: None,
+            state: None,
+            ..Default::default()
+        }
+    }
+}
+
+impl super::Request for Request {
+
+    type Proto = ListNetworkConnectionsArgs;
+
+    fn from_proto(proto: ListNetworkConnectionsArgs) -> Request {
+        Request {
+            listening_only: proto.listening_only.unwrap_or(false)
+        }
+    }
+}
+
+impl<'a> super::Response for Response<'a> {
+
+    const RDF_NAME: Option<&'static str> = Some("NetworkConnection");
+
+    type Proto = NetworkConnection;
+
+    fn into_proto(self) -> Self::Proto {
+        let mut result: NetworkConnection;
+        result = make_connection_from_socket_info(self.socket_info);
+        if let Some(process_info) = self.process_info {
+            result.pid = Some(process_info.pid);
+            result.process_name = process_info.name;
+        }
+        result
+    }
+}
+
+impl From<&Process> for ProcessInfo {
+
+    fn from(process: &Process) -> ProcessInfo {
+        ProcessInfo {
+            pid: process.pid() as u32,
+            name: Some(process.name().to_string())
+        }
+    }
+}
+
+impl ProcessInfo {
+
+    /// Constucts a ProcessInfo system from a given process ID. The process
+    /// name will not be set.
+    fn from_pid(pid: u32) -> ProcessInfo {
+        ProcessInfo {
+            pid,
+            name: None
+        }
+    }
+
+    /// Constucts a ProcessInfo system from a given process ID. The process
+    /// name will be retrieved from the system using `system` parameter.
+    fn from_system<S: SystemExt>(system: &S, pid: u32) -> ProcessInfo {
+        match system.get_process(pid as i32) {
+            Some(process) => ProcessInfo::from(process),
+            None => ProcessInfo::from_pid(pid)
+        }
+    }
+}
+
+/// Handles requests for the network connection action.
+pub fn handle<S: Session>(
+    session: &mut S,
+    request: Request
+) -> session::Result<()> {
+    let mut system = System::new();
+    system.refresh_processes();
+
+    let addr_family_flags = netstat2::AddressFamilyFlags::all();
+    // Optimization: when we are asked only for listening connections,
+    // we need only TCP protocol.
+    let protocol_flags = if request.listening_only {
+        netstat2::ProtocolFlags::TCP
+    } else {
+        netstat2::ProtocolFlags::all()
+    };
+
+    let connection_iter = netstat2::iterate_sockets_info(
+        addr_family_flags,
+        protocol_flags
+    );
+    let connection_iter = match connection_iter {
+        Ok(val) => val,
+        Err(err) => return Err(session::Error::action(Error::from(err)))
+    };
+
+    for connection in connection_iter {
+        let connection = match connection {
+            Ok(val) => val,
+            Err(err) => {
+                error!("Unable to get socket information: {}", err);
```

Comment:
So far I (and I think other projects as well) use lowercase sentences for log messages.